### PR TITLE
Add support for Container insights on Windows 

### PR DESCRIPTION
**Pull request by KlwntSingh**
**Description:** <Describe what has changed.>
This PR adds CI on Windows feature in awscontainerinsightsreceiver. This PR merges dev branch aws-ci-windows into aws-ci-dev

Ex. Adding a feature - Explain what this achieves.-->
This PR adds CI on Windows feature in awscontainerinsightsreceiver.

**Testing:** <Describe what testing was performed and which tests were added.>
1. Unit tests pass for Windows
```
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows      (cached)
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows/extractors   (cached)
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows/hcsshim      (cached)
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows/kubelet      (cached)
```
2. PR checks can confirm unit tests passing on Linux
3. Complete e2e testing passed.
Screenshot 2024-03-06 at 4 06 34 AM
Screenshot 2024-03-06 at 4 06 29 AM
Screenshot 2024-03-06 at 4 06 17 AM
Screenshot 2024-03-06 at 4 06 05 AM
Screenshot 2024-03-06 at 4 05 55 AM
Screenshot 2024-03-06 at 4 05 46 AM
Screenshot 2024-03-06 at 4 02 29 AM
Screenshot 2024-03-06 at 4 02 16 AM
Screenshot 2024-03-06 at 4 01 16 AM
Screenshot 2024-03-06 at 4 01 00 AM

4. Sample of output in performance log group for Windows node
```
{
    "AutoScalingGroupName": "eks-windows-2022-mng-42c6d6f1-7ce3-0ec3-XXXXXXXXXXXXXXX",
    "CloudWatchMetrics": [
        {
            "Namespace": "ContainerInsights",
            "Dimensions": [
                [
                    "ClusterName",
                    "InstanceId",
                    "NodeName"
                ],
                [
                    "ClusterName"
                ]
            ],
            "Metrics": [
                {
                    "Name": "node_status_condition_pid_pressure",
                    "Unit": "Count"
                },
                {
                    "Name": "node_status_condition_disk_pressure",
                    "Unit": "Count"
                },
                {
                    "Name": "node_status_condition_memory_pressure",
                    "Unit": "Count"
                },
                {
                    "Name": "node_cpu_reserved_capacity",
                    "Unit": "Percent"
                },
                {
                    "Name": "node_status_allocatable_pods",
                    "Unit": "Count"
                },
                {
                    "Name": "node_status_condition_ready",
                    "Unit": "Count"
                },
                {
                    "Name": "node_status_condition_unknown",
                    "Unit": "Count"
                },
                {
                    "Name": "node_number_of_running_pods",
                    "Unit": "Count"
                },
                {
                    "Name": "node_cpu_limit"
                },
                {
                    "Name": "node_status_capacity_pods",
                    "Unit": "Count"
                },
                {
                    "Name": "node_memory_limit",
                    "Unit": "Bytes"
                },
                {
                    "Name": "node_cpu_usage_total"
                },
                {
                    "Name": "node_memory_working_set",
                    "Unit": "Bytes"
                },
                {
                    "Name": "node_memory_reserved_capacity",
                    "Unit": "Percent"
                },
                {
                    "Name": "node_network_total_bytes",
                    "Unit": "Bytes/Second"
                },
                {
                    "Name": "node_cpu_utilization",
                    "Unit": "Percent"
                },
                {
                    "Name": "node_memory_utilization",
                    "Unit": "Percent"
                },
                {
                    "Name": "node_number_of_running_containers",
                    "Unit": "Count"
                }
            ]
        }
    ],
    "ClusterName": "XXXXX-CI-2022",
    "InstanceId": "i-XXXXXXXXXXXX",
    "InstanceType": "m5.large",
    "NodeName": "ip-XX-XX-XX-XX.us-west-2.compute.internal",
    "OperatingSystem": "windows",
    "Sources": [
        "kubelet",
        "pod",
        "calculated"
    ],
    "Timestamp": "1709726796214",
    "Type": "Node",
    "Version": "0",
    "kubernetes": {
        "host": "ip-XX-XX-XX-XX.us-west-2.compute.internal"
    },
    "node_cpu_request": 550,
    "node_cpu_usage_system": 0,
    "node_cpu_usage_user": 0,
    "node_memory_pgfault": 0,
    "node_memory_pgmajfault": 0,
    "node_memory_request": 448790528,
    "node_memory_rss": 0,
    "node_memory_usage": 2205806592,
    "node_network_rx_bytes": 324.75,
    "node_network_rx_dropped": 0,
    "node_network_rx_errors": 0,
    "node_network_tx_bytes": 207.86666666666667,
    "node_network_tx_dropped": 0,
    "node_network_tx_errors": 0,
    "node_cpu_limit": 2000,
    "node_cpu_reserved_capacity": 27.500000000000004,
    "node_cpu_usage_total": 2,
    "node_cpu_utilization": 0.1,
    "node_memory_limit": 8274866176,
    "node_memory_reserved_capacity": 5.423538199344531,
    "node_memory_utilization": 7.9183519112418335,
    "node_memory_working_set": 655233024,
    "node_network_total_bytes": 532.6166666666666,
    "node_number_of_running_containers": 2,
    "node_number_of_running_pods": 2,
    "node_status_allocatable_pods": 110,
    "node_status_capacity_pods": 110,
    "node_status_condition_disk_pressure": 0,
    "node_status_condition_memory_pressure": 0,
    "node_status_condition_pid_pressure": 0,
    "node_status_condition_ready": 1,
    "node_status_condition_unknown": 0
}
```
5. Link to Integ tests results for both Linux and Windows. CW agent runs successfully on Both Linux and Windows. Fluenbit fails due to timeout issue when destroying the cluster. There is no issue in functional working of fluentbit.
https://github.com/aws/private-amazon-cloudwatch-agent-staging/actions/runs/8175470234/job/22352686273

**Documentation:** <Describe the documentation added.>
This commit adds documentation - https://github.com/amazon-contributing/opentelemetry-collector-contrib/pull/166

### CODE REVIEW

File: receiver/awscontainerinsightreceiver/internal/k8swindows/k8swindows.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	k8sDecorator           stores.K8sDecorator
+	kubeletSummaryProvider *kubeletsummaryprovider.SummaryProvider
+	hcsshimStatsProvider   *hcsshim.HCSStatsProvider
+	hostInfo               host.Info
+	version                string
+}
+
+var metricsExtractors = []extractors.MetricExtractor{}
+
+func New(logger *zap.Logger, decorator *stores.K8sDecorator, hostInfo host.Info) (*K8sWindows, error) {
```
Comment (nathalapooja) [inline receiver/awscontainerinsightreceiver/internal/k8swindows/k8swindows.go:39]: There are no tests for this file methods

Comment (KlwntSingh) [reply receiver/awscontainerinsightreceiver/internal/k8swindows/k8swindows.go:37]: I added tests for this file. Unit test coverage is only limited to initializing k8sWindows object. 
Adding unit test for GetMetrics will require some refactoring. Can follow up with adding unit test ater release.

**Review by nathalapooja — COMMENTED** [review]
{
                    "Name": "node_cpu_limit"
                },
and
                {
                    "Name": "node_cpu_usage_total"
                },
metrics are missing units. May I know why didn't we include the statsd and emf integ testing for windows EKS?Have you added additional unit tests from yesterday discussion for Extractor helpers class?

**Review by mitali-salvi — APPROVED** [review]

**Comment by KlwntSingh** [comment]
> { "Name": "node_cpu_limit" }, and { "Name": "node_cpu_usage_total" }, metrics are missing units. May I know why didn't we include the statsd and emf integ testing for windows EKS?Have you added additional unit tests from yesterday discussion for Extractor helpers class?

 The following metrics are reported in unit of millicores, and cloudwatch doesn't support it.

**Pull request merged**